src/reducers/scrollStateHelper.js

```diff
@@ -252,8 +252,10 @@ export function updateRowCount(state, { rowsCount }) {
   state.scrollContentHeight = rowsCount * state.rowHeight;
   state.rowOffsets = PrefixIntervalTree.uniform(rowsCount, state.rowHeight);
   state.storedHeights = new Array(rowsCount);
+  state.rows = new Array(rowsCount);
   for (var i = 0; i < rowsCount; i++) {
     state.storedHeights[i] = state.rowHeight;
+    state.rows[i] = i;
```

Comment:
This seemed necessary for the client side filtering example to work... The problem being that when the data changes outside of FDT, the original row array was still being held onto, which meant BufferedRows would loop over the wrong set.